feat(wallet)_: Auto refresh token list

saledjenic opened the pull request:
This PR depends on:
- https://github.com/status-im/status-go/pull/6397
- https://github.com/status-im/status-go/pull/6398

Changes in this PR are not breaking changes and include the following:
- Fetching the list of tokens lists from remote, if the remote URL is not provided a local, hardcoded list will be used.
- Fetching the content of each remote list from the list of URLs (each URL points to a single tokens list).
- At this moment, updating the tokens lists is scheduled every 30 minutes, while the interval for checking for the update is set to every 3 minutes. The client should set the refresh interval, which will be added in one of the follow-up PRs.
- Old token stores are removed, and new, initial token lists, following the new form, are added.
- According to the previous point, scripts used by make commands `download-tokens` and `analyze-token-stores` are updated.
- The above-mentioned changes are integrated into `token.Manager` type.
- Each time tokens lists are update the `wallet.token-lists.updated` signal will be emitted

saledjenic committed:
feat(wallet)_: fetching list of token lists and fetching each token list from it added

saledjenic committed:
feat(wallet)_: tokens lists type for managing auto refresh tokens lists added

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (104)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/1/) | 2025-03-07 14:00:15 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250307-135719-00e797-pr6399.aar) |
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/1/) | 2025-03-07 14:00:17 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250307-135720-00e797-pr6399.zip) |
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/1/) | 2025-03-07 14:01:26 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250307-135723-00e797-pr6399.zip) |
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/1/) | 2025-03-07 14:03:10 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250307-135727-00e797-pr6399.zip) |
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/1/) | 2025-03-07 14:03:35 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250307-135724-00e797-pr6399.zip) |
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/1/) | 2025-03-07 14:03:45 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250307-135721-00e797-pr6399.zip) |
| :heavy_check_mark: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/1/) | 2025-03-07 14:16:46 | ~19 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/1//consoleText) |
| :heavy_multiplication_x: | 00e797db | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/1/) | 2025-03-07 14:29:22 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/2/) | 2025-03-07 14:04:06 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250307-140047-05c602-pr6399.aar) |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/2/) | 2025-03-07 14:04:18 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250307-140048-05c602-pr6399.zip) |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/2/) | 2025-03-07 14:06:45 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250307-140130-05c602-pr6399.zip) |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/2/) | 2025-03-07 14:06:54 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250307-140313-05c602-pr6399.zip) |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/2/) | 2025-03-07 14:08:59 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250307-140341-05c602-pr6399.zip) |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/2/) | 2025-03-07 14:09:19 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250307-140351-05c602-pr6399.zip) |
| :heavy_check_mark: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/2/) | 2025-03-07 14:33:02 | ~16 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/2//consoleText) |
| :heavy_multiplication_x: | 05c60266 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/2/) | 2025-03-07 14:59:31 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/2//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/3/) | 2025-03-07 21:52:56 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/3//consoleText) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/3/) | 2025-03-07 21:53:12 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250307-215018-e78b98-pr6399.aar) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/3/) | 2025-03-07 21:53:21 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250307-215020-e78b98-pr6399.zip) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/3/) | 2025-03-07 21:54:31 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250307-215027-e78b98-pr6399.zip) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/3/) | 2025-03-07 21:55:00 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250307-215024-e78b98-pr6399.zip) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/3/) | 2025-03-07 21:55:32 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250307-215024-e78b98-pr6399.zip) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/3/) | 2025-03-07 21:55:50 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250307-215021-e78b98-pr6399.zip) |
| :heavy_check_mark: | e78b98d0 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/3/) | 2025-03-07 22:05:56 | ~15 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/4/) | 2025-03-10 10:26:02 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250310-102306-7afb77-pr6399.aar) |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/4/) | 2025-03-10 10:26:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250310-102307-7afb77-pr6399.zip) |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/4/) | 2025-03-10 10:27:05 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250310-102317-7afb77-pr6399.zip) |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/4/) | 2025-03-10 10:27:45 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-102310-7afb77-pr6399.zip) |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/4/) | 2025-03-10 10:28:28 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250310-102309-7afb77-pr6399.zip) |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/4/) | 2025-03-10 10:29:25 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-102312-7afb77-pr6399.zip) |
| :heavy_check_mark: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/4/) | 2025-03-10 10:40:08 | ~16 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/4//consoleText) |
| :heavy_multiplication_x: | 7afb7734 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/4/) | 2025-03-10 10:55:05 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/5/) | 2025-03-10 12:39:19 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250310-123633-6ae05d-pr6399.zip) |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/5/) | 2025-03-10 12:39:36 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250310-123632-6ae05d-pr6399.aar) |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/5/) | 2025-03-10 12:40:31 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250310-123640-6ae05d-pr6399.zip) |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/5/) | 2025-03-10 12:41:18 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-123636-6ae05d-pr6399.zip) |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/5/) | 2025-03-10 12:41:50 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-123637-6ae05d-pr6399.zip) |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/5/) | 2025-03-10 12:42:23 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250310-123634-6ae05d-pr6399.zip) |
| :heavy_check_mark: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/5/) | 2025-03-10 12:53:58 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/5//consoleText) |
| :heavy_multiplication_x: | 6ae05dd1 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/5/) | 2025-03-10 13:06:56 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/6/) | 2025-03-10 13:31:23 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250310-132816-c7f107-pr6399.aar) |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/6/) | 2025-03-10 13:31:24 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250310-132818-c7f107-pr6399.zip) |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/6/) | 2025-03-10 13:32:21 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250310-132825-c7f107-pr6399.zip) |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/6/) | 2025-03-10 13:32:55 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-132821-c7f107-pr6399.zip) |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/6/) | 2025-03-10 13:33:36 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-132821-c7f107-pr6399.zip) |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/6/) | 2025-03-10 13:34:18 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250310-132819-c7f107-pr6399.zip) |
| :heavy_check_mark: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/6/) | 2025-03-10 13:45:22 | ~16 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/6//consoleText) |
| :heavy_multiplication_x: | c7f1074f | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/6/) | 2025-03-10 13:58:36 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/7/) | 2025-03-10 14:00:16 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250310-135727-a0f246-pr6399.zip) |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/7/) | 2025-03-10 14:00:38 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250310-135727-a0f246-pr6399.aar) |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/7/) | 2025-03-10 14:01:19 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-135730-a0f246-pr6399.zip) |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/7/) | 2025-03-10 14:01:59 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250310-135736-a0f246-pr6399.zip) |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/7/) | 2025-03-10 14:02:42 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250310-135728-a0f246-pr6399.zip) |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/7/) | 2025-03-10 14:04:19 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-135732-a0f246-pr6399.zip) |
| :heavy_check_mark: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/7/) | 2025-03-10 14:16:23 | ~18 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/7//consoleText) |
| :heavy_multiplication_x: | a0f246a7 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/7/) | 2025-03-10 14:29:15 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/8/) | 2025-03-10 14:20:16 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250310-141713-2d970f-pr6399.zip) |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/8/) | 2025-03-10 14:21:41 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250310-141712-2d970f-pr6399.aar) |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/8/) | 2025-03-10 14:21:50 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-141716-2d970f-pr6399.zip) |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/8/) | 2025-03-10 14:22:49 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250310-141720-2d970f-pr6399.zip) |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/8/) | 2025-03-10 14:23:21 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250310-141717-2d970f-pr6399.zip) |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/8/) | 2025-03-10 14:24:54 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250310-141715-2d970f-pr6399.zip) |
| :heavy_check_mark: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/8/) | 2025-03-10 14:39:07 | ~21 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/8//consoleText) |
| :heavy_multiplication_x: | 2d970fed | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/8/) | 2025-03-10 14:59:59 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/8//consoleText) |
| :heavy_multiplication_x: | 2d970fed | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/9/) | 2025-03-10 15:42:31 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/9/) | 2025-03-11 07:59:51 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250311-075655-7408f8-pr6399.aar) |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/9/) | 2025-03-11 08:00:03 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250311-075655-7408f8-pr6399.zip) |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/9/) | 2025-03-11 08:01:12 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250311-075703-7408f8-pr6399.zip) |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/9/) | 2025-03-11 08:01:36 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250311-075658-7408f8-pr6399.zip) |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/9/) | 2025-03-11 08:02:14 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250311-075659-7408f8-pr6399.zip) |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/9/) | 2025-03-11 08:03:10 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250311-075657-7408f8-pr6399.zip) |
| :heavy_check_mark: | 7408f841 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/9/) | 2025-03-11 08:14:02 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/9//consoleText) |
| :heavy_multiplication_x: | 7408f841 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/10/) | 2025-03-11 08:27:59 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/10//consoleText) |
| :heavy_multiplication_x: | 7408f841 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/11/) | 2025-03-11 09:09:54 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/10/) | 2025-03-14 07:44:27 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250314-074128-77acac-pr6399.zip) |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/10/) | 2025-03-14 07:44:35 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250314-074127-77acac-pr6399.aar) |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/10/) | 2025-03-14 07:45:46 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250314-074135-77acac-pr6399.zip) |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/10/) | 2025-03-14 07:46:03 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250314-074131-77acac-pr6399.zip) |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/10/) | 2025-03-14 07:46:41 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250314-074132-77acac-pr6399.zip) |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/10/) | 2025-03-14 07:47:43 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250314-074129-77acac-pr6399.zip) |
| :heavy_check_mark: | 77acac46 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/10/) | 2025-03-14 08:01:57 | ~20 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/10//consoleText) |
| :heavy_multiplication_x: | 77acac46 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/12/) | 2025-03-14 08:15:43 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/12//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/11/) | 2025-03-14 12:33:58 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250314-123052-056eba-pr6399.aar) |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/11/) | 2025-03-14 12:34:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250314-123053-056eba-pr6399.zip) |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/11/) | 2025-03-14 12:35:25 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250314-123100-056eba-pr6399.zip) |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/11/) | 2025-03-14 12:35:35 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250314-123056-056eba-pr6399.zip) |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/11/) | 2025-03-14 12:36:12 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250314-123056-056eba-pr6399.zip) |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/11/) | 2025-03-14 12:36:52 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250314-123055-056eba-pr6399.zip) |
| :heavy_check_mark: | 056ebafe | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/11/) | 2025-03-14 12:52:10 | ~21 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/11//consoleText) |
| :heavy_check_mark: | 056ebafe | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/13/) | 2025-03-14 13:07:00 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/12/) | 2025-03-14 13:47:41 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250314-134439-bce20b-pr6399.zip) |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/12/) | 2025-03-14 13:47:50 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250314-134438-bce20b-pr6399.aar) |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/12/) | 2025-03-14 13:48:44 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250314-134446-bce20b-pr6399.zip) |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/12/) | 2025-03-14 13:49:22 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250314-134442-bce20b-pr6399.zip) |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/12/) | 2025-03-14 13:49:57 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250314-134442-bce20b-pr6399.zip) |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/12/) | 2025-03-14 13:50:41 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250314-134440-bce20b-pr6399.zip) |
| :heavy_check_mark: | bce20bde | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/12/) | 2025-03-14 14:05:12 | ~20 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/12//consoleText) |
| :heavy_check_mark: | bce20bde | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/14/) | 2025-03-14 14:19:10 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/14//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 29920dee | [#13](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/13/) | 2025-03-17 16:54:34 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250317-164946-29920d-pr6399.zip) |
| :heavy_check_mark: | 29920dee | [#13](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/13/) | 2025-03-17 16:55:28 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250317-164941-29920d-pr6399.zip) |
| :heavy_check_mark: | 29920dee | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/13/) | 2025-03-17 17:11:12 | ~21 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250317-164936-29920d-pr6399.aar) |
| :heavy_check_mark: | 29920dee | [#13](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/13/) | 2025-03-17 17:12:22 | ~22 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250317-164941-29920d-pr6399.zip) |
| :heavy_check_mark: | 29920dee | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/13/) | 2025-03-17 17:14:17 | ~24 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250317-164939-29920d-pr6399.zip) |
| :heavy_check_mark: | 29920dee | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/15/) | 2025-03-17 17:43:09 | ~53 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/15//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/14/) | 2025-03-17 22:14:39 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250317-221200-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/14/) | 2025-03-17 22:14:54 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250317-221159-e3a12d-pr6399.aar) |
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/14/) | 2025-03-17 22:15:37 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250317-221203-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/14/) | 2025-03-17 22:16:19 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250317-221206-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/14/) | 2025-03-17 22:17:54 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250317-221203-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/14/) | 2025-03-17 22:18:20 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250317-221201-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/14/) | 2025-03-17 22:31:08 | ~19 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/14//consoleText) |
| :heavy_multiplication_x: | e3a12db5 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/16/) | 2025-03-17 22:46:23 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/16//consoleText) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/15/) | 2025-03-18 07:11:04 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250318-070804-e3a12d-pr6399.aar) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/15/) | 2025-03-18 07:11:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250318-070806-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/15/) | 2025-03-18 07:11:47 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250318-070809-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/15/) | 2025-03-18 07:13:01 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250318-070813-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/15/) | 2025-03-18 07:13:47 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250318-070809-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/15/) | 2025-03-18 07:14:21 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250318-070806-e3a12d-pr6399.zip) |
| :heavy_check_mark: | e3a12db5 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/15/) | 2025-03-18 07:25:42 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/15//consoleText) |
| :heavy_multiplication_x: | e3a12db5 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/17/) | 2025-03-18 07:42:49 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/17//consoleText) |
| | | | | | | |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6399/16/) | 2025-03-18 09:22:13 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250318-091932-dc0c7a-pr6399.zip) |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6399/16/) | 2025-03-18 09:22:30 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250318-091930-dc0c7a-pr6399.aar) |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6399/16/) | 2025-03-18 09:23:12 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250318-091934-dc0c7a-pr6399.zip) |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6399/16/) | 2025-03-18 09:24:05 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250318-091940-dc0c7a-pr6399.zip) |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6399/16/) | 2025-03-18 09:25:15 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250318-091936-dc0c7a-pr6399.zip) |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6399/16/) | 2025-03-18 09:25:46 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250318-091933-dc0c7a-pr6399.zip) |
| :heavy_check_mark: | dc0c7a3f | [#16](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/16/) | 2025-03-18 09:39:29 | ~19 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6399/16//consoleText) |
| :heavy_check_mark: | dc0c7a3f | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/18/) | 2025-03-18 09:54:21 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6399/18//consoleText) |

saledjenic committed:
chore(wallet)_: initial tokens lists preparation

downloader and analyzer moved to more appropriate location and accordingly updated to the changes made before.

`make download-tokens` should be used to update initial tokens lists for the app.

`make analyze-token-stores` should be used to analyze initial tokens lists.

saledjenic committed:
chore(wallet)_: integrating tokens list into tokens package

Old token stores are not needed anymore since the tokens lists
are being fetched dynamically and maintained differently.

friofry commented on services/wallet/token/token-lists/token_lists.go:
```diff
@@ -0,0 +1,168 @@
+// If the remote list url is not set (empty string provided), the hardcoded default list will be used.
+// The auto-refresh interval is used to fetch the list of token lists from the remote source and update the local cache.
+// The auto-refresh check interval is used to check if the auto-refresh should be triggered.
+func (t *TokenLists) Start(remoteListUrl string, autoRefreshInterval time.Duration, autoRefreshCheckInterval time.Duration) {
```
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it, - [documentation for context](https://pkg.go.dev/context#:~:text=Do%20not%20store%20Contexts%20inside%20a%20struct%20type)


https://zealous-polka-dc7.notion.site/Code-smell-2-Pass-context-to-functions-cd49904001ca4152be006124e64f5c61

> This approach isn't ideal because contexts in Go are meant to flow down from the top (entry point) of the application. Each component typically receives a context parameter from its caller, rather than generating a new background context internally. Ideally, we would refactor so that: Add a context parameter to the wallet.Start() function. This ensures the caller explicitly controls the lifecycle and cancellation of routines.
Eventually, we might(should) manage a single "root context" at the StatusNode (get_status_node.go) level (the top-level entry point). This would allow coordinated cancellation of all services (30 services at the moment) at shutdown.

saledjenic replied:
@friofry I agree, but that requires refactoring of the upper code. If I just remove it from the struct I won't be able to cancel the context on calling `Stop()`. I've added "todo" and will try to address that in the follow-up PR.

friofry commented on services/wallet/token/token-lists/fetcher/fetcher.go:
```diff
@@ -0,0 +1,71 @@
+
+	for _, tokenList := range tokenLists {
+		group.Add(func(c context.Context) error {
+			fetchTokenList(tokenList, tokenChannel)
```
context is not passed to `fetchTokenList`, which means the HTTP request can't be canceled.

saledjenic replied:
Updated.

friofry commented on services/wallet/token/token-lists/fetcher/fetcher.go:
```diff
@@ -0,0 +1,71 @@
+
+	for _, tokenList := range tokenLists {
+		group.Add(func(c context.Context) error {
+			fetchTokenList(tokenList, tokenChannel)
```
Maybe it makes sense to skip the notification if all `fetchTokenList` calls failed?
Currently the notification is sent regardless of whether any token lists were successfully fetched and stored. Error handling in goroutines doesn't propagate errors back to the main function.

First I was thinking to suggest returning error from `fetchTokenList`, but it seems that the behavior of `NewAtomicGroup` Add method is confusing. If the functor returns error, the whole group is canceled.

saledjenic replied:
> Maybe it makes sense to skip the notification if all fetchTokenList calls failed?

Maybe for discussion, but notification is sent in the last line in this function via `notify` channel. It should be sent even when fetching fails for all lists. In the UI we display the info "Last fetch ..." so I guess that label should be updated so that the user knows when was the last try and the next one will be in X minutes.

> First I was thinking to suggest returning error from fetchTokenList, but it seems that the behavior of NewAtomicGroup Add method is confusing. If the functor returns error, the whole group is canceled.

We don't return errors cause even if an error happens for call X, we still want to do call X+1, X+2...

friofry replied:
> so that the user knows when was the last try 

I think it's a bad UI pattern to show "synchronized just now" when the data hasn't actually been synchronized.

I was thinking of something like that:
```
   for fetchedList := range tokenChannel {
       if err := t.storeTokenList(fetchedList.ID, fetchedList.JsonData); err != nil {
           logutils.ZapLogger().Error("Failed to store token list", zap.Error(err))
       } else {
           successCount++
       }
   }
   
   if successCount > 0 {
       notify <- struct{}{}
   }
   ```

friofry commented on services/wallet/token/token-lists/fetcher/constants.go:
```diff
@@ -0,0 +1,48 @@
+  {
+    "id": "uniswap",
+    "sourceUrl": "https://ipfs.io/ipns/tokens.uniswap.org",
+    "schema": "https://uniswap.org/tokenlist.schema.json"
```
is this a valid URL ?
```
 curl -X GET https://ipfs.io/ipns/tokens.uniswap.org
curl: (35) Recv failure: Operation timed out
```

saledjenic replied:
I don't see any issues here. Also, trying this command returns an instant response on my side.

friofry replied:
We have many users in different countries.
Probably making this data available from our market proxy will allow us to solve this problem.

friofry commented on services/wallet/token/token-lists/default-lists/downloader/main.go:
```diff
@@ -0,0 +1,120 @@
+}
+
+func downloadTokens(client *http.Client, key string, source defaulttokenlists.TokensSource) {
+	response, err := client.Get(source.SourceURL)
```
Token list sounds like a big chunk of data that rarely change.
I suggest optimizing it:

* store etag header for the downloaded token list. And do not download what we already have. (`If-None-Match:` or `If-Modified-Since` and do not download if returns 304)
* Introduction of context cancellation of HTTP request 
* Addition of connection and request timeouts against hanging requests
```
func fetchContentWithCaching(ctx context.Context, url string, etag string) ([]byte, string, bool, error) {
    req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
    if err != nil {
        return nil, "", false, err
    }
    
    // ETag 
    if etag != "" {
        req.Header.Add("If-None-Match", etag)
    }
    
    // Timeouts
    transport := &http.Transport{
	DialContext: (&net.Dialer{
		Timeout: dialTimeout, // Timeout for establishing a connection
	}).DialContext,
	TLSHandshakeTimeout:   tlsHandshakeTimeout,   // Timeout for TLS handshake
	ResponseHeaderTimeout: responseHeaderTimeout, // Timeout for receiving response headers
    }

    client := &http.Client{
        Transport: transport,
        Timeout: requesetTimeout * time.Second,
    }
    
    resp, err := client.Do(req)
    if err != nil {
        return nil, "", false, err
    }
    defer resp.Body.Close()
    
    // HTTP 304 Not Modified
    if resp.StatusCode == http.StatusNotModified {
        return nil, etag, true, nil
    }
    
    // Content changed
    body, err := ioutil.ReadAll(resp.Body)
    if err != nil {
        return nil, "", false, err
    }
    
    // update ETag
    newEtag := resp.Header.Get("ETag")
    if newEtag == "" {
        newEtag = etag // if etag is missing
    }
    
    return body, newEtag, false, nil
}
```

saledjenic replied:
I like this idea. Will do it today in the follow-up PR.

Actually, I see now that we have `thirdparty.NewHTTPClientWithDetailedTimeouts` and will extend it with the new function `func fetchContentWithCaching(ctx context.Context, url string, etag string) ([]byte, string, bool, error)` and use it from there in a new PR.

friofry replied:
Cool! It would be great to use the same http client

friofry commented on services/wallet/token/token-lists/refresher.go:
```diff
@@ -0,0 +1,56 @@
+				if time.Since(lastTokensUpdate) < autoRefreshInterval {
+					continue
+				}
+
```
Maybe storing the `lastTokensUpdated` after the FetchAndNotify will be successful?

saledjenic replied:
`FetchAndNotify` doesn't return any error, it fetches the lists, stores the content of each list and notifies. `LastTokensUpdate` when the app tried to fetch the lists and is updated even when fetching fails for all lists.

friofry replied:
I don't have a strong opinion on this. I just feel that without request timeouts and small autoRefreshInterval it might work unexpectedly (it's 30 * time.minute in token.go, but 3 * time.second in tests).

saledjenic commented:
@friofry thanks for the review and good comments. The last commit is about the changes you proposed. Once you agree I will move the code to appropriate commits, but this way is easier for you to review. 

I will handle context and etag in the follow-up PRs.

friofry commented:
> @friofry thanks for the review and good comments. The last commit is about the changes you proposed. Once you agree I will move the code to appropriate commits, but this way is easier for you to review.
> 
> I will handle context and etag in the follow-up PRs.

Sounds like a good plan to make these changes as a separate PR.

friofry reviewed (approved):
Looks good to me, please add me as a reviewer for the follow-up PRs on the context and etag

saledjenic commented:
@friofry here is a PR  https://github.com/status-im/status-go/pull/6433 addressing the things we've discussed here about.
